addon/mixins/pikaday.js

```diff
@@ -99,17 +104,31 @@ export default Ember.Mixin.create({
   },
 
   setMinDate: function() {
-    if (this.get('minDate')) {
-      run.later(() => {
-        this.get('pikaday').setMinDate(this.get('minDate'));
+    const { enforceDateIntervals, pikaday, minDate, value } = getProperties(this, [ 'enforceDateIntervals', 'pikaday', 'minDate', 'value' ]);
+
+    if (minDate) {
+      run.later( () => {
+        pikaday.setMinDate(minDate);
+
+        // Force current value to not be lower than minDate
+        if ( enforceDateIntervals && value < minDate) {
+          get(this, 'onSelection')(minDate);
+        }
       });
     }
   },
 
   setMaxDate: function() {
-    if (this.get('maxDate')) {
-      run.later(() => {
-        this.get('pikaday').setMaxDate(this.get('maxDate'));
+    const { enforceDateIntervals, pikaday, maxDate, value }  = getProperties(this, [ 'enforceDateIntervals', 'pikaday', 'maxDate', 'value' ]);
+
+    if (maxDate) {
+      run.later( () => {
+        pikaday.setMaxDate(maxDate);
+
+        // Force current value to not be greated than maxDate
+        if ( enforceDateIntervals && value > maxDate ) {
+          get(this, 'onSelection')(maxDate);
```

Comment:
Following DDUA pattern I think it's the passed function the one that should take care of setting the value.